src/tests/ftest/launch.py

```diff
@@ -1406,12 +1407,25 @@ def get_yaml_data(yaml_file):
         dict: the contents of the yaml file
 
     """
+    class DaosLoader(yaml.SafeLoader):  # pylint: disable=too-many-ancestors
```

Comment:
The only concern would be making sure this change doesn't cause a problem with any existing yaml files.  Running the `pr` tagged tests will cover most of the cases, but we would want to verify compliance with a test that uses `!mux` in the `test_servers:` assignment, like https://github.com/daos-stack/daos/blob/master/src/tests/ftest/erasurecode/online_rebuild_mdtest.yaml.  The `test_ec_online_rebuild_mdtest` has a `pr` tag so this should be covered.  

Running the following command manually should verify this will not break any pr, daily, or weekly testing:
```
./launch.py -l -m -d -n auto -ts wolf-[A-B] pr daily_regression full_regression
```
(`wolf-[A-B]` does not need to be a large list of host - 2 hosts with at least 1 NVMe each should suffice.)

Reply:
Seems to work correctly from all my testing, and I added a print to the return of this function to be sure.  This is only loading the yaml files into a python datatype so is a fairly self-contained change.

Here are the results for that test, it has three variants as expected https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-7286/52/testReport/FTEST_erasurecode/EcodOnlineRebuildMdtest/